File: go-client-sdk/pkg/models/operations/getcompaniescompanyiddatainvoicesinvoiceid.go
package operations

import (
	"github.com/speakeasy-sdks/codat-sdks/go-client-sdk/v2/pkg/models/shared"
)

type GetCompaniesCompanyIDDataInvoicesInvoiceIDPathParams struct {
	CompanyID string `pathParam:"style=simple,explode=false,name=companyId"`
	InvoiceID string `pathParam:"style=simple,explode=false,name=invoiceId"`
}

type GetCompaniesCompanyIDDataInvoicesInvoiceIDSecurity struct {
	APIKeyAuth *shared.SchemeAPIKeyAuth `security:"scheme,type=apiKey,subtype=header"`
	CodatLogin *shared.SchemeCodatLogin `security:"scheme,type=oauth2"`
}

type GetCompaniesCompanyIDDataInvoicesInvoiceIDRequest struct {
	PathParams GetCompaniesCompanyIDDataInvoicesInvoiceIDPathParams
	Security   GetCompaniesCompanyIDDataInvoicesInvoiceIDSecurity
}

type GetCompaniesCompanyIDDataInvoicesInvoiceIDResponse struct {
	CodatDataContractsDatasetsInvoice *shared.CodatDataContractsDatasetsInvoice
	ContentType                       string
	StatusCode                        int
}

File: typescript-client-sdk/src/sdk/models/operations/getcompaniescompanyiddatainvoicesinvoiceid.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import * as shared from "../shared";


export class GetCompaniesCompanyIdDataInvoicesInvoiceIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=companyId" })
  companyId: string;

  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=invoiceId" })
  invoiceId: string;
}

export class GetCompaniesCompanyIdDataInvoicesInvoiceIdSecurity extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "security, scheme=true;type=apiKey;subtype=header" })
  apiKeyAuth?: shared.SchemeAPIKeyAuth;

  @SpeakeasyMetadata({ data: "security, scheme=true;type=oauth2" })
  codatLogin?: shared.SchemeCodatLogin;
}

export class GetCompaniesCompanyIdDataInvoicesInvoiceIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  pathParams: GetCompaniesCompanyIdDataInvoicesInvoiceIdPathParams;

  @SpeakeasyMetadata()
  security: GetCompaniesCompanyIdDataInvoicesInvoiceIdSecurity;
}

export class GetCompaniesCompanyIdDataInvoicesInvoiceIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  codatDataContractsDatasetsInvoice?: shared.CodatDataContractsDatasetsInvoice;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;
}